aptos-move/aptos-vm-benchmarks/src/main.rs

```diff
@@ -0,0 +1,219 @@
+    let dirs = read_dir(path.as_path()).unwrap();
+
+    //// Setting up local execution environment once
+    // disable parallel execution
+    let executor = FakeExecutor::from_head_genesis();
+    let mut executor = executor.set_not_parallel();
+
+    //// Go over all Move projects
+    for dir in dirs {
+        let entry = dir.unwrap();
+        if !entry.path().is_dir() {
+            continue;
+        }
+        let dir_path = entry.path();
+
+        let build_options = BuildOptions {
+            with_srcs: true,
+            with_abis: true,
+            with_source_maps: true,
+            with_error_map: true,
+            ..BuildOptions::default()
+        };
+        let package =
+            BuiltPackage::build(dir_path, build_options).expect("build package must succeed");
+
+        let codes = package.extract_code();
+        for code in codes {
+            // avoid SEQUENCE_NUMBER_TOO_NEW error
+            let mut sequence_num_counter = 0;
+
+            let compiled_module = CompiledModule::deserialize(&code).unwrap();
+
+            // get module address
+            let mut module_bytes = vec![];
+            compiled_module.serialize(&mut module_bytes).unwrap();
+            let module_id = compiled_module.self_id();
+            let address = &module_id.address();
+            let identifier = &module_id.name().as_str();
+
+            // find non-entry functions and ignore them
+            // keep entry function names in func_identifiers vector
+            let funcs = compiled_module.function_defs;
+            let func_handles = compiled_module.function_handles;
+            let func_identifier_pool = compiled_module.identifiers;
+
+            // find # of params in each func if it is entry function
+            let signature_pool = compiled_module.signatures;
+
+            let mut func_identifiers = Vec::new();
+            for func in funcs {
+                let is_entry = func.is_entry;
+                if !is_entry {
+                    continue;
+                }
+
+                let func_idx: usize = func.function.0.into();
+                let handle = &func_handles[func_idx];
+
+                let func_identifier_idx: usize = handle.name.0.into();
+                let func_identifier = &func_identifier_pool[func_identifier_idx];
+
+                // skip if it doesn't start with "benchmark"
+                let func_name = func_identifier.as_str();
+                if !func_name.starts_with(PREFIX) {
+                    continue;
+                }
+
+                // skip if it doesn't match pattern
+                let mut fully_qualified_path = String::new();
+                fully_qualified_path.push_str(address.to_string().as_str());
+                fully_qualified_path.push_str("::");
+                fully_qualified_path.push_str(identifier);
+                fully_qualified_path.push_str("::");
+                fully_qualified_path.push_str(func_name);
```

Comment:
You can replace this with `format!("{}::{}::{}", address, identifier, func_name)`